File: 2IMN15/web/scripts/domain/User2Device.js
function User2Device(userID, deviceID, prioLevel, red, green, blue, lowLight) {
    var self = this;

    self.userID = userID;
    self.deviceID = deviceID;
    self.prioLevel = prioLevel;
    self.red = red;
    self.green = green;
    self.blue = blue;
    self.lowLight = lowLight;



}

var parseUser2Device = function(data) {
    return new User2Device(data.userID, data.deviceID, data.prioLevel, data.red, data.green, data.blue, data.lowLight);
}

File: 2IMN15/src/iot/domain/User2Device.java
package iot.domain;

import iot.Validation;


public class User2Device {
    private int userID;
    private int deviceID;
    private int prioLevel;
    private int red;
    private int green;
    private int blue;
    private boolean lowLight;


    /**
     * Initializes a new instance of the UserAccount class.
     */
    private User2Device() { }

    public int getUserID() {
        return this.userID;
    }
    public int getDeviceID() {
        return this.deviceID;
    }
    public int getPrioLevel() {
        return this.prioLevel;
    }
    public int getRed() {
        return this.red;
    }
    public int getGreen() {
        return this.green;
    }
    public int getBlue() {
        return this.blue;
    }
    public boolean getLowLight() {
        return this.lowLight;
    }




    public static User2Device Make(
            int userID,
            int deviceID,
            int prioLevel,
            int red,
            int green,
            int blue,
            boolean lowLight

    ) {
        Validation.userExist(userID);
        Validation.deviceID(deviceID);
        Validation.priolevel(prioLevel);
        Validation.color(red);
        Validation.color(green);
        Validation.color(blue);


        User2Device u2d = new User2Device();
        u2d.userID = userID;
        u2d.deviceID = deviceID;
        u2d.prioLevel = prioLevel;
        u2d.red = red;
        u2d.green = green;
        u2d.blue = blue;
        u2d.lowLight = lowLight;


        return u2d;
    }
}
